lisa/notifiers/perfdump.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import json
+import os
+from dataclasses import dataclass
+from datetime import datetime
+from decimal import Decimal
+from enum import Enum
+from typing import Any, List, Type, cast
+
+from dataclasses_json import dataclass_json
+
+from lisa import constants, messages, notifier, schema
+
+
+@dataclass_json()
+@dataclass
+class PerfDumpSchema(schema.Notifier):
+    path: str = "lisa.json"
+
+
+class PerfDump(notifier.Notifier):
+    """
+    The Json notifier is used to output the perf test results in JSON format.
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "perfdump"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return PerfDumpSchema
+
+    def _received_message(self, message: messages.MessageBase) -> None:
+        if isinstance(message, messages.PerfMessage):
+            message_dict = {}
+            for key, value in message.__dict__.items():
+                if isinstance(value, Enum):
+                    value = value.value
+                elif isinstance(value, datetime) or isinstance(value, Decimal):
+                    value = str(value)
+                message_dict[key] = value
+
+            json.dump(message_dict, self._report_file, indent=4, ensure_ascii=False)
+            self._report_file.write(",")
```

Comment:
The tailing `,` may not be loaded by some libraries. It's better to append `,` before each new line.

Reply:
updated